src/internal/components/pluggableVisualizations/geoChart/tests/PluggableGeoPushpinChart.spec.tsx

```diff
@@ -44,5 +44,99 @@ describe("PluggableGeoPushpinChart", () => {
                 expect(extendedReferencePoint.uiConfig).toEqual(uiConfigMocks.defaultGeoPushpinUiConfig);
             });
         });
+
+        it("should transform view by attribute to location attribute", async () => {
+            const { oneMetricAndCategoryAndStackReferencePoint } = referencePointMocks;
+            const geoAttribute: IBucketItem = {
+                aggregation: null,
+                attribute: "geo.attr",
+                localIdentifier: "a1",
+                type: "attribute",
+                isLocationIconVisible: true,
+            };
+            // update view by attribute from normal attribute to geo attribute
+            oneMetricAndCategoryAndStackReferencePoint.buckets[1].items = [geoAttribute];
```

Comment:
This mutation to `oneMetricAndCategoryAndStackReferencePoint ` could affect other tests which depends on it.